### PR TITLE
Add method to reset SupportsArrayTranslateTRxx flag

### PR DESCRIPTION
**Pull request by dchopra001**
Signed-off-by: Dhruv Chopra <Dhruv.C.Chopra@ibm.com>

### CODE REVIEW

File: compiler/z/codegen/OMRCodeGenerator.cpp [hunk 1]
```diff
@@ -490,7 +490,7 @@ OMR::Z::CodeGenerator::CodeGenerator()
       }
    self()->setSupportsArrayCmp();
    self()->setSupportsArrayCmpSign();
-   if (!comp->compileRelocatableCode())
+   if (!(comp->compileRelocatableCode() || comp->isOutOfProcessCompilation()))
```
Comment (fjeremic) [inline compiler/z/codegen/OMRCodeGenerator.cpp]: This `isOutOfProcessCompilation` only seems to exist in OpenJ9 as far as I can tell. See Travis build failure. Do we need to overload in OMR?

Comment (dchopra001) [reply compiler/z/codegen/OMRCodeGenerator.cpp]: I think we'll need a different query here (I'm not sure what to use yet). I should have disabled `ci` tests from running on this and opened this as a draft PR.

**Comment by dchopra001** [comment]
@fjeremic I've add a query here to toggle array translate support. We can override this query in OpenJ9 so that support can be enabled/disabled accordingly. The OpenJ9 PR would look like this: https://github.com/dchopra001/openj9/commit/3ba355e2df20de240a1d2206343ef6954753eba1.

Please take a look. If this is approved/merged, then I'll open the corresponding OpenJ9 PR as well.

We should be able to do this without breaking anything.

**Comment by fjeremic** [comment]
> Please take a look. If this is approved/merged, then I'll open the corresponding OpenJ9 PR as well.

I question the need for this code in OMR. Why can't we just check for remote compiles in OpenJ9 in the `J9::CodeGenerator` constructor and disable it there?

**Comment by dchopra001** [comment]
> I question the need for this code in OMR. Why can't we just check for remote compiles in OpenJ9 in the J9::CodeGenerator constructor and disable it there?

I've added a `resetSupportsArrayTranslateTRxx` routine here. I'll add a check in a corresponding openj9 PR that will invoke this routine to disable support if we detect that a remote compile is being performed.

**Comment by fjeremic** [comment]
@genie-omr build all

**Pull request merged**